common/v2/services/Store/Account/AccountProvider.tsx

```diff
@@ -32,6 +34,23 @@ export class AccountProvider extends Component {
       service.updateAccount(uuid, accountData);
       this.getAccounts();
     },
+    addNewTransactionToAccount: (accountData, newTransaction) => {
+      const existingTransaction = accountData.transactions.find(
+        tx => tx.hash === newTransaction.hash
+      );
+      delete newTransaction.network;
+      if (existingTransaction) {
+        const newTransactionSet = accountData.transactions.filter(
+          transaction => transaction.hash !== newTransaction.hash
+        );
+        newTransactionSet.push(newTransaction);
+        accountData.transactions = [...newTransactionSet];
+      } else {
+        accountData.transactions.push(newTransaction);
+      }
+      service.updateAccount(accountData.uuid, accountData);
+      this.getAccounts();
+    },
```

Comment:
Since it would be better to be using Immutable objects (it eliminates a whole category of bugs), we can actually be creating a new object instead of mutating the original one:
```
addNewTransactionToAccount: (accountData, newTransaction) => {
   const { uuid, transactions } = accountData;
   const { network, ...txWithoutNetwork } = newTransaction;
   const account = {
      ....accountData,
      transactions: [
          ...transactions.filter(tx => tx.hash === newTransaction.hash),  // will return all elements if no match found
          txWithoutNetwork
      ]
   }

   service.updateAccount(uuid, account);
   this.getAccounts();
}
```

Reply:
nice spot